tests/RobotFramework/tests/cumulocity/registration/device_registration.robot

```diff
@@ -166,19 +165,26 @@ Entities persisted and restored
     Execute Command    tedge mqtt pub --retain 'te/factory/shop/plc1/metrics' '{"@type":"service","@id":"plc1-metrics","@parent":"factory/shop/plc1/"}'
     Execute Command    tedge mqtt pub --retain 'te/factory/shop/plc2/metrics' '{"@type":"service","@id":"plc2-metrics","@parent":"factory/shop/plc2/"}'
 
-    Should Have MQTT Messages    c8y/s/us    message_contains=101,plc1    date_from=${timestamp}    minimum=1    maximum=1
-    Should Have MQTT Messages    c8y/s/us    message_contains=101,plc2    date_from=${timestamp}    minimum=1    maximum=1
-    Should Have MQTT Messages    c8y/s/us/plc1    message_contains=101,plc1-sensor1    date_from=${timestamp}    minimum=1    maximum=1
-    Should Have MQTT Messages    c8y/s/us/plc1    message_contains=101,plc1-sensor2    date_from=${timestamp}    minimum=1    maximum=1
-    Should Have MQTT Messages    c8y/s/us/plc2    message_contains=101,plc2-sensor1    date_from=${timestamp}    minimum=1    maximum=1
-    Should Have MQTT Messages    c8y/s/us/plc1    message_contains=102,plc1-metrics    date_from=${timestamp}    minimum=1    maximum=1
-    Should Have MQTT Messages    c8y/s/us/plc2    message_contains=102,plc2-metrics    date_from=${timestamp}    minimum=1    maximum=1
+    External Identity Should Exist    plc1
+    External Identity Should Exist    plc2
+    External Identity Should Exist    plc1-sensor1
+    External Identity Should Exist    plc1-sensor2
+    External Identity Should Exist    plc2-sensor1
+    External Identity Should Exist    plc1-metrics
+    External Identity Should Exist    plc2-metrics
```

Comment:
Even a simple 1s sleep here would have solved the issue. But, doing a deterministic assertion here looked better.